File: CS/ExecuteSprocWorkflowExample.Module/CustomNativeActivity2.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Workflow.Activities;
using System;
using System.Activities;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ExecuteSprocWorkflowExample.Module {
    public class CustomNativeActivity2 : NativeActivity {
        protected override void Execute(NativeActivityContext context) {
            XPObjectSpace xpObjectSpace = this.XPObjectSpace.Get(context);
            xpObjectSpace.Session.ExecuteSproc("StoredProcName");
        }
        [DefaultValue(null)]
        public InArgument<XPObjectSpace> XPObjectSpace { get; set; }
    }
}

File: VB/ExecuteSprocWorkflowExample.Module/CustomNativeActivity2.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.Workflow.Activities
Imports System
Imports System.Activities
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks

Namespace ExecuteSprocWorkflowExample.Module
    Public Class CustomNativeActivity2
        Inherits NativeActivity

        Protected Overrides Sub Execute(ByVal context As NativeActivityContext)

            Dim xpObjectSpace_Renamed As XPObjectSpace = Me.XPObjectSpace.Get(context)
            xpObjectSpace_Renamed.Session.ExecuteSproc("StoredProcName")
        End Sub
        <System.ComponentModel.DefaultValue(CType(Nothing, Object))> _
        Public Property XPObjectSpace() As InArgument(Of XPObjectSpace)
    End Class
End Namespace
